### PR TITLE
[Reviewer: Matt] Monit file changes

### PR DESCRIPTION
**Pull request by eleanor-merry**
Matt, can you review these changes to the chronos monit file? 

The changes:
* add cpu/memory checks
* stop the poll_chronos script from running if chronos isn't
* add a restart command (that doesn't abort the process)

If you're happy, then I'll make the same set of changes to the other monit files

### CODE REVIEW

File: usr/share/chronos/chronos.monit [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+# Check the HTTP interface. This depends on the Chronos process (and so won't run
+# unless the Chronos process is running)
 check program poll_chronos with path "/usr/share/clearwater/bin/poll_chronos.sh"
+  depends on chronos
 
-  if status != 0 for 2 cycles
-     then exec "/etc/init.d/chronos abort"
+  # Aborting generates a core file and triggers diagnostic collection. Monit will raise
+  # an alarm when it restarts the process
+  if status != 0 for 2 cycles then exec "/bin/bash -c '/usr/share/chronos/write_monit_restart_diags; /etc/init.d/chronos abort'"
```
Comment (mirw) [inline usr/share/chronos/chronos.monit]: Does "write_monit_restart_diags" already exist?  It doesn't seem to be referenced by existing code.  Should it be triggered by "chronos abort"?

File: usr/share/chronos/chronos.monit [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 
+  # The start, stop and restart commands are linked to alarms
   start program = "/bin/bash -c '/usr/share/clearwater/bin/issue_alarm.py monit 3000.4; /etc/init.d/chronos start'"
   stop program  = "/bin/bash -c '/usr/share/clearwater/bin/issue_alarm.py monit 3000.4; /etc/init.d/chronos stop'"
+  restart program = "/bin/bash -c '/usr/share/clearwater/bin/issue_alarm.py monit 3000.4; /etc/init.d/chronos restart'"
 
+  # Check the service's resource usage, and abort the process if it's too high. This will
+  # generate a core file and trigger diagnostics collection. Monit will raise an alarm when
+  # it restarts the process
+  if cpu > 80% for 6 cycles then exec "etc/init.d/chronos abort"
```
Comment (mirw) [inline usr/share/chronos/chronos.monit]: Why do we not write_monit_restart_diags here?

File: usr/share/chronos/chronos.monit (hunk 1 above)
Comment (eleanor-merry) [inline usr/share/chronos/chronos.monit]: It does - it used to be called by the chronos.monit script before it was removed in https://github.com/Metaswitch/chronos/commit/fcc837dd56eef8b1c1b060336c8c451f29706377

File: usr/share/chronos/chronos.monit (hunk 2 above)
Comment (eleanor-merry) [inline usr/share/chronos/chronos.monit]: It's not particularly useful to. The script provides diagnostics to help if monit was loses track of Chronos

**Comment by mirw** [comment]
Fine.

**Pull request merged**